photon-client/src/main/scala/com/linkedin/photon/ml/data/avro/AvroDataReader.scala

```diff
@@ -335,21 +354,36 @@ object AvroDataReader {
     .headOption
 
   /**
+   * Determines whether all the types are numeric
+   *
+   * @param types the types to check
+   * @return true if all specified types are numeric
+   */
+  protected[data] def allNumericTypes(types: Seq[Schema.Type]): Boolean =
+    types.forall(numericPrecedence.toSet)
+
+  /**
+   * Selects the "dominant" avro numeric type from the list. Dominance in this sense means the numeric type with highest
+   * precedence.
+   *
+   * @param types the avro types from which to select
+   * @return the dominant numeric type
+   */
+  protected[data] def getDominantNumericType(types: Seq[Schema.Type]): Schema.Type =
+    numericPrecedence.filter(types.toSet).head
+
+  /**
    * Converts the named avro field schema to an equivalent spark sql schema type.
    *
    * @param name The field name
    * @param avroSchema The avro schema for the field
    * @return Spark sql schema for the field
    */
   protected[data] def avroTypeToSql(name: String, avroSchema: Schema): Option[StructField] =
-
     avroSchema.getType match {
-      case INT => Some(StructField(name, IntegerType, nullable = false))
-      case STRING => Some(StructField(name, StringType, nullable = false))
-      case BOOLEAN => Some(StructField(name, BooleanType, nullable = false))
-      case DOUBLE => Some(StructField(name, DoubleType, nullable = false))
-      case FLOAT => Some(StructField(name, FloatType, nullable = false))
-      case LONG => Some(StructField(name, LongType, nullable = false))
+      case avroType @ (INT | LONG | FLOAT | DOUBLE | STRING | BOOLEAN) =>
```

Comment:
One issue though is that now a bit costs us 64 bits potentially :-/ Is that possibly a problem?